later/lib/database/datasources/records_local_data_source.dart

```diff
@@ -36,13 +35,49 @@ class RecordsLocalDataSource extends RecordsDataSource {
     return record;
   }
 
+  @override
+  Future<RecordDbModel> update(RecordDbModel recordDbModel) async {
+    final dbModel = await _isar.recordDbModels.get(recordDbModel.id);
+
+    if (dbModel != null) {
+      dbModel
+        ..title = recordDbModel.title == dbModel.title
+            ? dbModel.title
+            : recordDbModel.title
+        ..url =
+            recordDbModel.url == dbModel.url ? dbModel.url : recordDbModel.url
+        ..description = recordDbModel.description == dbModel.description
+            ? dbModel.description
+            : recordDbModel.description
+        ..lastEditedAt = DateTime.now();
+    } else {
+      throw const FormatException(
+          'We can not update file that we don\'t have at Database.');
+    }
+
+    await _isar.writeTxn(() => _isar.recordDbModels.put(dbModel));
+
+    return dbModel;
+  }
+
   @override
   Future<RecordDbModel?> getById(int id) {
-    return _isar.txn(() => _isar.recordDbModels.get(id));
+    final model = _isar.txn(() => _isar.recordDbModels.get(id));
+
+    if (model == null) {
+      throw const FormatException('Object with that id don\'t exist.');
```

Comment:
doesn't 😅

Reply:
Another thing: `FormatException` is usually related to format problems, such as incorrect title format or sth. In this case, it is not a format problem, it's a database problem. We'll cover this topic in the future, for now just return a null.